packages/perseus-editor/src/components/graph-settings.tsx

```diff
@@ -28,70 +28,111 @@ const defaultBackgroundImage = {
 function numSteps(range: any, step: any) {
     return Math.floor((range[1] - range[0]) / step);
 }
-
-const GraphSettings = createReactClass({
-    displayName: "GraphSettings",
-
-    propTypes: {
-        ...Changeable.propTypes,
-        editableSettings: PropTypes.arrayOf(
-            PropTypes.oneOf(["canvas", "graph", "snap", "image", "measure"]),
-        ),
-        box: PropTypes.arrayOf(PropTypes.number),
-        labels: PropTypes.arrayOf(PropTypes.string),
-        range: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.number)),
-        step: PropTypes.arrayOf(PropTypes.number),
-        gridStep: PropTypes.arrayOf(PropTypes.number),
-        snapStep: PropTypes.arrayOf(PropTypes.number),
-        valid: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
-        backgroundImage: PropTypes.object,
-        markings: PropTypes.oneOf(["graph", "grid", "none"]),
-        showProtractor: PropTypes.bool,
-        showRuler: PropTypes.bool,
-        showTooltips: PropTypes.bool,
-        rulerLabel: PropTypes.string,
-        rulerTicks: PropTypes.number,
-    },
-
-    getDefaultProps: function () {
-        return {
-            editableSettings: ["graph", "snap", "image", "measure"],
-            box: [
-                interactiveSizes.defaultBoxSizeSmall,
-                interactiveSizes.defaultBoxSizeSmall,
-            ],
-            labels: ["x", "y"],
-            range: [
-                [-10, 10],
-                [-10, 10],
-            ],
-            step: [1, 1],
-            gridStep: [1, 1],
-            snapStep: [1, 1],
-            valid: true,
-            backgroundImage: defaultBackgroundImage,
-            markings: "graph",
-            showProtractor: false,
-            showRuler: false,
-            showTooltips: false,
-            rulerLabel: "",
-            rulerTicks: 10,
-        };
-    },
-
-    getInitialState: function () {
+type Props = {
+    editableSettings: any[];
```

Comment:
The prototype looks like this was just `ReadonlyArray<"canvas" | "graph" | "snap" | "image" | "measure">`.  Can you try swapping `any` for that?